doc/design.md

```diff
@@ -0,0 +1,111 @@
+# rust-td design
+
+The rust-td is one design choice for a TD.
+
+A `td-shim` takes over the reset vector and prepares the TD environemnt setup. It could be considered as a lightweigth TD Virtual Firmware (TDVF). The td-shim will transfer control to a td-payload.
+
+A `td-payload` is a new execution environment. It could be a bare-metal environment, a UEFI virtual firmware, an OS loader, a OS kernel, etc.
+
+Below figure shows the high level design.
+
+   ```
+                     +----------------------+
+                     |       td-payload     |
+                     |                      |
+                     +----------------------+
+                                 ^
+                                 | <-------------------- td-shim spec 
+                     +----------------------+
+                     |        td-shim       |
+                     |           ^          |
+                     |           |          |
+                     |     (reset vector)   |
+      Guest          +----------------------+
+      ============================================ <--- td-shim spec
+      Host           +----------------------+
+                     |         VMM          |
+                     +----------------------+
+
+   ```
+
+ The [td-shim specification](https://github.com/jyao1/td-shim/blob/init_version/doc/tdshim_spec.md) defines the interface between td-shim and vmm, and the interface between td-shim and td-payload.
+
+ The [td-shim threat model](https://github.com/jyao1/td-shim/blob/init_version/doc/threat_model.md) defines the threat model for the td-shim.
+
+This repo includes a full `td-shim`, and sample `td-payload`. The consumer may create other td-payload. For example, to support [TDX](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-trust-domain-extensions.html) [migration TD](https://www.intel.com/content/dam/develop/external/us/en/documents/tdx-migration-td-design-guide-348987-001.pdf), a rust-migtd can inlcude a td-shim and a migtd-payload.
+
+## td-shim
+
+[rust-tdshim](https://github.com/jyao1/td-shim/tree/init_version/rust-tdshim) is a core of td-shim. The entrypoint is `_start()` at [main](https://github.com/jyao1/td-shim/blob/init_version/rust-tdshim/src/main.rs). It will initialize the td-shim and switch to td-payload at `switch_stack_call()` of [main](https://github.com/jyao1/td-shim/blob/init_version/rust-tdshim/src/main.rs).

```

Comment:
Do the URL need to be fixed to point to `confidential-containers` instead of `jyao1`?

Reply:
Oh. Right. Definitely. I will fix all before final push.
Here I point to the branch for easy review.

Reply:
All links are fixed to confidential-containers.